python/ray/dashboard/client/src/pages/dashboard/tune/Tune.tsx

```diff
@@ -83,8 +113,89 @@ class Tune extends React.Component<
     });
   };
 
+  handleExperimentChange = (event: React.ChangeEvent<{ value: any }>) => {
+    this.setState({
+      experiment: event.target.value,
+    });
+  };
+
+  handleExperimentSubmit = () => {
+    this.setState({ loading: true });
+    try {
+      setTuneExperiment(this.state.experiment)
+        .then(() => {
+          window.clearTimeout(this.timeout);
+          this.refreshTuneInfo().then(() => {
+            this.setState({ loading: false });
+          });
+        })
+        .catch((e) => {
+          this.props.setError(e.toString());
+          this.setState({ loading: false });
+        });
```

Comment:
Let's make this an `async` function and use `await` syntax here instead of a promise. Then we can also avoid duplicating the `catch` block.